.github/workflows/release.yml

```diff
@@ -1,45 +1,96 @@
-name: Release agent-kilt
+name: release
 
 on:
-  push:
-    tags:
-      - 'v*'
+  workflow_dispatch:
+    inputs:
+      release_version:
+        description: 'The release version, e.g. 5.0.0.'
+        type: string
+        required: true
 
-env:
-  GO_VERSION: 1.21.x
+      rc_number:
+        description: 'The release candidate number, e.g. 1.'
+        type: string
+        required: true
+
+      serverless_agent_version:
+        description: 'The version of the serverless-agent to be referenced by this release, e.g., 4.0.0. It will use the same version as the release if not specified.'
+        type: string
+        required: false
 
 jobs:
   release:
+    env:
+      RC_NAME: ${{ inputs.release_version }}-rc${{ inputs.rc_number }}
     name: Release
     runs-on: ubuntu-latest
     steps:
-      - name: Set up Go
-        uses: actions/setup-go@v4
-        with:
-          go-version: ${{ env.GO_VERSION }}
-        id: go
+      - name: Verify inputs
+        shell: bash
+        run: |
+          if [[ ! "${{ inputs.release_version }}" =~ ^([0-9]+)\.([0-9]+)\.([0-9]+)$ ]]; then
+              echo "The provided release version is not valid"
+              exit 1
+          fi
+
+          if [[ ! "${{ inputs.rc_number }}" =~ ^([0-9])+$ ]]; then
+              echo "The provided rc tag is not valid"
+              exit 1
+          fi
+
+          docker pull "quay.io/sysdig/workload-agent:${{ inputs.serverless_agent_version }}" &> /dev/null
+          exit_code=$?
+          if [[ $exit_code -ne 0 ]]; then
+            echo "The provided serverless_agent_version does not exists."
+            exit 1
+          fi
 
       - name: Checkout Code
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Build agent-kilt handler
+      - name: Push RC tag
+        shell: bash
+        run: |
+          git fetch --tags
+          if git rev-parse "${RC_NAME}" &> /dev/null; then
+            echo "The tag ${RC_NAME} already exists"
+            exit 1
+          fi
+          git tag "${RC_NAME}"
+          git push -u origin "${RC_NAME}"
+
+      - name: Prepare dist package
+        shell: bash
+        run: |
+          rm -rf dist
+          mkdir -p dist/provisioning/cloudformation
+
+      - name: Add versioned CloudFormation templates
+        env:
+          RELEASE_VERSION: ${{ inputs.release_version }}
+          SERVERLESS_AGENT_VERSION: ${{ inputs.serverless_agent_version || inputs.release_version }}
+        shell: bash
         run: |
-          make -C runtimes/cloudformation clean cmd/handler/handler
+          script="s/(dev)/${RELEASE_VERSION}/g; s/agent:latest/${SERVERLESS_AGENT_VERSION}/g"
```

Comment:
Considering the second sed. Will this not geneate the following:
```
INPUT
workload-agent:latest

OUTPUT
workload-4.0.0
```

Should the image tag not include the RC number?

Reply:
Nice spot :big-facepalm:

Reply:
> Should the image tag not include the RC number?

That's the same approach we currently have in place. It allows QE to verify the default values and avoids (re)versioning the templates during promotion. Plus, since it's just the default parameter value, QE can easily tweak it when deploying the template.

Also, the agent RCs aren't hosted on quay anymore, and I'd rather not expose the internal schema.

Reply:
Fixed, thank you

I squashed the fix commit https://github.com/sysdiglabs/agent-kilt/compare/12114df52eacdac15a37af8aef41e30b56930d63..26738d6b22346554bde070a4e255d8eb88fc1b9e